ibis/backends/flink/ddl.py

```diff
@@ -54,15 +58,31 @@ def _format_watermark_strategy(watermark: Watermark) -> str:
 
 
 def format_schema_with_watermark(
-    schema: sch.Schema, watermark: Watermark | None = None
+    schema: sch.Schema,
+    watermark: Watermark | None = None,
+    primary_keys: set[str] | None = None,
```

Comment:
I would say `primary_key`; there's really just one `primary_key`, but it can be compound.

Also, does it makes sense to also accept a `str`? I feel like at least `str | Iterable[str]` is more aligned with what other things that accept a key take (e.g. `group_by`).

Reply:
Thanks. I changed the argument to `primary_key`, which is a `str` that contains a single column name or a comma-separated list of column names.

Reply:
I agree with @deepyaman here (see my comment above). This should be something like:

```suggestion
    primary_keys: str | Sequence[str] | None = None,
```

with consuming code below updated accordingly. You might find `ibis.util.promote_list` useful here.

Reply:
list or set?